src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -325,6 +326,26 @@ MTR_AVAILABLE(ios(16.1), macos(13.0), watchos(9.1), tvos(16.1))
  */
 - (void)removeClientDataForKey:(NSString *)key endpointID:(NSNumber *)endpointID MTR_UNSTABLE_API;
 
+/**
+ * Download log of the desired type from the device.
+ *
+ * Note: The consumer of this API should move the file that the url points to or open it for reading before the
+ * completion handler returns. Otherwise, the file will be deleted, and the data will be lost.
+ *
+ * @param type       The type of log being requested. This should correspond to a value in the enum MTRDiagnosticLogType.
+ * @param timeout    The timeout for getting the log. If the timeout expires, completion will be called with whatever
+ *                   has been retrieved by that point (which might be none or a partial log).
+ *                   If the timeout is set to 0, the request will not expire and completion will not be called until
+ *                   the log is fully retrieved or an error occurs.
+ * @param queue      The queue on which completion will be called.
+ * @param completion The completion that will be called to return the URL of the requested log if successful. Otherwise
+ *                   returns an error.
```

Comment:
Does something like the following would be more understandable ?
'''
@param completion The completion handler that is called after attempting to retrieve the requested log. 
                                   - If there are no logs available, it returns a nil URL and a nil error.
                                   - In cases of errors where no logs are available, it returns a nil URL and a non-nil error.
                                   - If log retrieval is successful and logs are available, it returns a non-nil URL and a nil error.
                                   - If an error occurs but parts of the logs has been downloaded, it returns a non-nil URL and a non-nil error.
'''

I can also update the code to always returns an url if there is no error.

|  I don't think we should be making this look like a Swift async method if it's not following the rules for those...
What would you like it to look ?

Reply:
I have updated the completion values with: 
```
* @param completion The completion handler that is called after attempting to retrieve the requested log.
 *                     - In case of success, the completion handler is called with a non-nil URL and a nil error.
 *                     - If there is an error, a non-nil error is used and the url can be non-nil too if some logs has already been downloaded.
 *
```